Content.Server/Bed/Cryostorage/CryostorageSystem.cs

```diff
@@ -301,6 +301,11 @@ public override void Update(float frameTime)
             if (Timing.CurTime < containedComp.GracePeriodEndTime)
                 continue;
 
+            if (containedComp.Cryostorage == null ||
```

Comment:
Should this not be checking that the mind's attached entity is the cryostored entity? Im not sure this check would actually fix the issue. The body still needs to get removed, but the mind just needs to not detach.